framework/test/unit/modules/dpos_v2/method.spec.ts

```diff
@@ -140,4 +147,190 @@ describe('DposModuleApi', () => {
+			tokenIDFee: Buffer.from('0000000000000000', 'hex'),
+			delegateRegistrationFee: BigInt(10) * BigInt(10) ** BigInt(8),
+			maxBFTWeightCap: 500,
+			commissionIncreasePeriod: COMMISSION_INCREASE_PERIOD,
+			maxCommissionIncreaseRate: MAX_COMMISSION_INCREASE_RATE,
+		};
+		let tokenMethod: any;
+
+		beforeEach(async () => {
+			tokenMethod = { lock: jest.fn() };
+			dposMethod.init(moduleName, defaultConfig, tokenMethod);
+			await delegateSubStore.set(createStoreGetter(stateStore), address1, delegateData1);
+			await delegateSubStore.set(createStoreGetter(stateStore), address2, delegateData2);
+			await delegateSubStore.set(createStoreGetter(stateStore), address3, delegateData3);
+			jest.spyOn(tokenMethod, 'lock');
+			jest.spyOn(delegateSubStore, 'set');
+		});
+
+		it('should return if totalVotesReceived is 0', async () => {
+			await expect(
+				dposMethod.updateSharedRewards(methodContext, address1, tokenID1, BigInt(50)),
+			).resolves.toBeUndefined();
+			expect(tokenMethod.lock).not.toHaveBeenCalled();
+			expect(delegateSubStore.set).not.toHaveBeenCalled();
+		});
+
+		it('should initialize sharing coefficient to zero and set the appropriate amounts to delegate store for the specified token if there does not exist an item in delegateStore for the token id', async () => {
+			const newTokenID = utils.getRandomBytes(TOKEN_ID_LENGTH);
+			const reward = BigInt(50);
+			const rewardQ = q96(reward);
+			const commissionQ = q96(BigInt(delegateData2.commission));
+			const rewardFractionQ = q96(BigInt(1)).sub(commissionQ.div(q96(BigInt(10000))));
+			const selfVotesQ = q96(delegateData2.selfVotes);
+			const totalVotesQ = q96(delegateData2.totalVotesReceived);
+			const oldSharingCoefficient = q96(0);
+			const sharingCoefficientIncrease = rewardQ.muldiv(rewardFractionQ, totalVotesQ);
+			const sharedRewards = sharingCoefficientIncrease.mul(totalVotesQ.sub(selfVotesQ)).floor();
+			const newSharingCoefficient = oldSharingCoefficient.add(sharingCoefficientIncrease);
+			const updatedDelegateData = delegateData2;
+			updatedDelegateData.sharingCoefficients.push({
+				tokenID: newTokenID,
+				coefficient: newSharingCoefficient.toBuffer(),
+			});
+
+			await expect(
+				dposMethod.updateSharedRewards(methodContext, address2, newTokenID, reward),
+			).resolves.toBeUndefined();
+			expect(tokenMethod.lock).toHaveBeenCalledTimes(1);
+			expect(delegateSubStore.set).toHaveBeenCalledTimes(1);
+			expect(tokenMethod.lock).toHaveBeenCalledWith(
+				methodContext,
+				address2,
+				moduleName,
+				newTokenID,
+				sharedRewards,
+			);
+			expect(delegateSubStore.set).toHaveBeenCalledWith(
+				methodContext,
+				address2,
+				updatedDelegateData,
+			);
+		});
+
+		it('should lock the appropriate amount of rewards and update the delegate store in correct order with updated sharing coefficients if the token id is aready present but sharing coefficients are not sorted by token id', async () => {
```

Comment:
> token id is aready present but sharing coefficients are not sorted

I think this shouldnt happen because it will be sorted every time it is mutated

Reply:
what is mutated ? `delegateData3` is not used before.

Reply:
What I meant was you can assume delegate store sharing coefficient are always sorted. so in delegate 3, the sharing coefficient is not sorted, but this wont happen.
in the logic, you are pushing the new sharing coefficient at the end, and this will make the sharing coefficient unsorted, and the logic is sorting before setting the value at the end

Reply:
I think there is some confusion interpreting the LIP logic here. 

Since the sorting is indented "one tab space inside" the `if` condition in the LIP, "sorting" should be done ONLY if the item with token id does not exist. 

> in the logic, you are pushing the new sharing coefficient at the end, and this will make the sharing coefficient unsorted, and the logic is sorting before setting the value at the end

In the current implementation however, I am sorting ALWAYS irrespective of the presence of the item and AFTER pushing the new element so there is no chance of unsorted coefficients like you mention.

If you agree that my LIP interpretation is valid I'd move the sorting logic inside the `if` logic.

Reply:
I think the source code is correct, the point is we always have sorted coefficient.

My point was in the test setup for `delegate3`, it shouldn't have empty / unsorted coefficient

Reply:
> in the logic, you are pushing the new sharing coefficient at the end, and this will make the sharing coefficient unsorted, and the logic is sorting before setting the value at the end
What do you mean by this ?